swag/utils.py

```diff
@@ -282,47 +350,81 @@ async def update_forbes_classement(guild, swag_client, client):
+
+    # Subdivision de la liste du classement en sous-liste de taille équitable
+    forbes_chunks = list(chunks(forbes, ligne_par_message))
+
+    # Subdivision de la liste des €agnotte
+    cagnottes_chunks = list(chunks(cagnottes, ligne_par_message))
 
     # Récupération du nombre de message nécessaire pour écrire tout le
-    # classement (c'est le nombre de sous-listes)
-    nbr_pages = math.ceil(len(forbes) / personne_par_message)
+    # forbes (c'est le nombre de sous-listes totaux)
+    nbr_pages_classement = math.ceil(len(forbes) / ligne_par_message)
+    nbr_pages_cagnottes = math.ceil(len(cagnottes) / ligne_par_message)
+    nbr_pages = nbr_pages_classement + nbr_pages_cagnottes
 
     # On compte le nombre de message posté dans le $wag forbes
     nbr_message_in_channel = 0
     async for message in channel_forbes.history(oldest_first=True):
         nbr_message_in_channel += 1
 
     # Si le nombre de message du canal est plus petit que le nombre
-    # de messages nécessaire pour écrire le classement on en créé
+    # de messages nécessaire pour écrire le classement et les cagnottes on en créé
     for _ in range(nbr_pages - nbr_message_in_channel):
-        await channel_forbes.send("Nouvelle page de classement en cours d'écriture")
+        await channel_forbes.send("Nouvelle page du forbes en cours d'écriture")
 
-    # édition des messages pour mettre à jour le classement
-    cpt_message = 0
+    # édition des messages pour mettre à jour le forbes
+    cpt_message_classement = 0
+    cpt_message_cagnottes = 0
     async for message in channel_forbes.history(oldest_first=True):
-        await message.edit(
-            content=await mini_forbes_swag(
-                forbes_chunks[cpt_message], cpt_message + 1, guild, client
+
+        if (
+            cpt_message_cagnottes < nbr_pages_cagnottes
+        ):  # On écrit d'abord les €agnottes
+            await message.edit(
+                content=await mini_forbes_cagnottes(
+                    cagnottes_chunks[cpt_message_cagnottes],
+                    guild,
+                    client,
+                )
             )
-        )
-        cpt_message += 1
+            cpt_message_cagnottes += 1
+
+        else:  # Ensuite, le classement
+
+            await message.edit(
+                content=await mini_forbes_swag(
+                    forbes_chunks[cpt_message_classement],
+                    cpt_message_classement + 1,
+                    guild,
+                    client,
+                )
+            )
+            cpt_message_classement += 1
 
     # update des bonus de st$le
     swag_client.swag_bank.update_bonus_growth_rate()
     # update du rôle du "Bobby $wag"
     await update_the_swaggest(guild, swag_client)
 
 
+def currency_to_str(enum_currency: Currency):
+    if enum_currency == 0:
+        return "$wag"
+    elif enum_currency == 1:
+        return "$tyle"
+
+
 def forbes_medal(rank):
     if rank == 1:
         return "🥇"
```

Comment:
Some french remains in this file but other than that it's ok

Reply:
Resolved in : https://github.com/BobbySoftwares/Bybbot/pull/7/commits/c9cfecdb424780b7243e2b2ba8fcc0fba8a01e8f